app/assets/javascripts/govuk_publishing_components/analytics-ga4/gtm-page-views.js

```diff
@@ -0,0 +1,46 @@
+;(function (global) {
+  'use strict'
+
+  var GOVUK = global.GOVUK || {}
+
+  GOVUK.Gtm = {
+    sendPageView: function () {
+      if (window.dataLayer) {
+        var data = {
+          event: 'config_ready',
+          page: {
+            location: this.getLocation(),
+            referrer: this.getReferrer(),
+            title: this.getTitle(),
+            status_code: this.getStatusCode()
+          }
+        }
```

Comment:
Sorry to keep banging the drum for a generic GA4 data structure, I feel it would be best to create such a beast though - that way it's consistent throughout - less cognitive load on the devs and it'll really help the data teams with the YoY/UA mapping issue too.

Currently, we have this...
```
gtm_attributes: {
  event: ‘analytics’,
  event_name: 
  link_url:
  ui: {
    type: TYPE OF ELEMENT OR COMPONENT,
    text: NAME OR TITLE OF THE SPECIFIC ELEMENT,
    index: INDEX OF THE SPECIFIC ELEMENT, e.g. 1 (of 4) (optional),
    'index-total': TOTAL NUMBER OF ELEMENTS IN THE PARENT, e.g. 4 (optional),
    section: PART OF THE PAGE INTERACTED WITH (optional),
    state: STATE OF THE ELEMENT WHEN INTERACTED e.g. ‘opened’ (optional),
  }
}
```

If we change this structure slightly (and update the existing code of course) then map in these attributes  something like this...

```
data: {
  event: ‘config_ready’,
  event_name:  this.getTitle(),
  url: this.getLocation(),
  referrer: this.getReferrer(),
  status: this.getStatusCode(),
  state: {
    type: 'pageload',
    text: 'n/a',
    index: 'n/a',
    'index-total': 'n/a',
    section: 'n/a',
  }
}
```
So, we it doesn't matter what we call it, but we keep a consistent structure inside it and `n/a` anything that is not relevant for that specific interaction type.

Reply:
Having a consistent data structure is good, but apparently the analysts are happy with the structure as it is because a page view is a distinct thing from other kinds of events (even though they're all events).

And yes, there could be empty parts to the data, but they should always have `n/a` values.